standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -294,7 +294,7 @@ static void prewarm(RawStore rawStore) {
               AggrStats aggrStatsAllButDefaultPartition = null;
               if (table.isSetPartitionKeys()) {
                 Deadline.startTimer("getPartitions");
-                partitions = rawStore.getPartitions(catName, dbName, tblName, Integer.MAX_VALUE);
+                partitions = rawStore.getPartitions(catName, dbName, tblName, -1);
```

Comment:
Why is this change needed? It looks like it doesn't have to do with anything we wanted to achieve in this PR. Same on 623 line below.

Reply:
Most of these changes have been backported. I haven't added them manually. I've added them with `git cherry-pick`.